modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/i18nHelper.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+interface ILocalizedItemDetails {
+	rootPropertyName: string;
+	value: string;
+	valuePath: Array<string>;
+}
+
+const languageId = Liferay.ThemeDisplay.getLanguageId();
+const BCP47LanguageId = Liferay.ThemeDisplay.getBCP47LanguageId();
+const defaultLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
+
+function getLanguageKey(data: any): string {
+	let languageKey = defaultLanguageId as string;
+
+	if (data[languageId]) {
+		languageKey = languageId as string;
+	}
+	else if (data[BCP47LanguageId]) {
+		languageKey = BCP47LanguageId;
+	}
+
+	return languageKey;
+}
+
+export function getLocalizedValueFromItem(
```

Comment:
I think `FromItem` suffix doesn't really add to descriptiveness of utility. Can we call this just `getLocalizedValue`? This seems slightly better than `getLocalizedLabel`.

By [convention](https://github.com/liferay/liferay-frontend-projects/blob/master/guidelines/general/file_names.md), file name should be the same as exported function, so `getLocalizedValue.ts`.

Reply:
With the file name I was thinking in possible future extensions, but it absolutely makes sense if exported individually. :ok_hand:

Reply:
Single function per file is also good for Jest tests, where we can have multiple tests per one function, in a single tests file. This is more relevant for global utilities, where we are more strict with unit tests coverage.

Reply:
Although.. should this be a global utility? :thinking: Is the logic tied to FDS, or applicable to any place that uses Objects endpoints?

Reply:
If it's a true global utility, it should be in `frontend-js-web`.

Reply:
> Although.. should this be a global utility? 🤔 Is the logic tied to FDS, or applicable to any place that uses Objects endpoints?

Right now, I think that it is tied to FDS. Also from our conversations, I created it as a local utility, making it globally in the future if we find other use cases. 
Anyway, we could move to `frontend-js-web` if it fits better

Reply:
>  I created it as a local utility, making it globally in the future if we find other use cases.

 :+1: